src/libxtp/ERIs.cc

```diff
@@ -192,7 +192,116 @@ namespace votca {
         }
         
         
-        
+		
+		
+		void ERIs::CalculateERIs_4c_direct(const AOBasis& dftbasis, const ub::matrix<double> &DMAT)
+		{
+			cout << endl;
+			cout << "ERIS.cc ERIs::CalculateERIs_4c_direct" << endl;
+			
+			// Initialize ERIs matrix
+			_ERIs = ub::zero_matrix<double>(DMAT.size1(), DMAT.size2());
+			// Get number of shells
+			int numShells = dftbasis.getNumofShells();
+			
+			// Test
+			printf("size(DMAT) = [%d, %d]\n", int(DMAT.size1()), int(DMAT.size2()));
+			printf("dftbasisSize = %d\n", dftbasis.AOBasisSize());
+			printf("numShells = %d\n", numShells);
+
+			// Timer
+			clock_t start = clock();
+
+			#pragma omp parallel for
+			for (int iShell_3 = 0; iShell_3 < numShells; iShell_3++)
+			{
+				const AOShell* shell_3 = dftbasis.getShell(iShell_3); // Shell object
+				int start_3 = shell_3->getStartIndex(); // Basis function index offset
+				int numFunc_3 = shell_3->getNumFunc(); // Number of basis functions in the shell
+
+				for (int iShell_4 = 0; iShell_4 < numShells; iShell_4++)
+				{
+					const AOShell* shell_4 = dftbasis.getShell(iShell_4);
+					int start_4 = shell_4->getStartIndex();
+					int numFunc_4 = shell_4->getNumFunc();
+
+					for (int iShell_1 = 0; iShell_1 < numShells; iShell_1++)
+					{
+						const AOShell* shell_1 = dftbasis.getShell(iShell_1);
+						int start_1 = shell_1->getStartIndex();
+						int numFunc_1 = shell_1->getNumFunc();
+
+						for (int iShell_2 = 0; iShell_2 < numShells; iShell_2++)
```

Comment:
you iterate over all combinations of i,j,k,l instead due to symmetry you should only iterate over 1/8 of it and then add the result to all respective combinations.